src/fosslight_scanner/fosslight_scanner.py

```diff
@@ -255,60 +263,114 @@ def init(output_path=""):
     return os.path.isdir(_output_dir), output_root_dir, result_log
 
 
+def check_comapre_output_file(output_file_name, format):
```

Comment:
It's a minor issue, so it's okay to review it later. 🙂

Reply:
I created check_compare_output_file in run_compare at first. But for the log output directory, it needs to be called before init().

Reply:
`Init() `is called in multiple places, how about making it only called once in `run_main`?
Because there are quite a few parts that call logger in `run_main `before `init `is called.

Reply:
Then CUSTOMIZED_FORMAT_FOR_COMPARE_MODE will be in run_main. So I wish to keep it.

Reply:
OK. 👌